kotlinx-coroutines-core/native/src/CoroutineContext.kt

```diff
@@ -8,33 +8,29 @@ import kotlinx.coroutines.internal.*
 import kotlin.coroutines.*
 import kotlin.native.concurrent.*
 
-private fun takeEventLoop(): EventLoopImpl =
-    ThreadLocalEventLoop.currentOrNull() as? EventLoopImpl ?:
-        error("There is no event loop. Use runBlocking { ... } to start one.")
-
 internal actual object DefaultExecutor : CoroutineDispatcher(), Delay {
+
+    private val delegate = WorkerDispatcher(name = "Dispatchers.Default")
```

Comment:
Maybe the following would work?
```suggestion
    private val delegate by lazy { newSingleThreadContext(name = "Dispatchers.Default") }
```
Then we could make `WorkerDispatcher` private, and also be loudly notified if we accessed this in a single-threaded configuration.

Reply:
`delegate` has to expose `Delay` supertype, so it won't work

Reply:
Whereas just doing `as Delay` where needed is undesirable because of the runtime checks?

Reply:
In K/N runtime performance is not really something we keep in mind for now, `WorkerDispatcher` approach just looks more straightforward